### PR TITLE
feature#26 added an autocomplete input using google place api

### PR DESCRIPTION
**Pull request by marwan011**

### CODE REVIEW

**Comment by marwan011** [comment]
Added inputs (start and destination) using google place api (autocomplete and place details).
The autocomplete merge data between outdoor and indoor location.
-Indoor will return : x,y coordinate and a room identificator
-Outdoor will return : latitude, longitude

File: App.tsx [hunk 1]
```diff
@@ -118,12 +133,8 @@ class App extends Component<{}, appState> {
       );
     }
     return (
-      <View>
-        <Text>
-          indoor Component (should have a button somewhere that change the displayIndoor state to
-          false when user is done whit indoor)
-        </Text>
-      </View>
+     // <Text>inddor component </Text>
```
Comment (razine-bensari) [inline App.tsx]: comment code should be removed

File: components/AutoCompleteInput.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    zIndex:8,
+    width: 250,
+    height: 45,
+    borderColor: Colors.black,
+    backgroundColor: Colors.white,
+    borderWidth: 0.5,
+    padding: 5,
+}
+}
+);
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: indentation should be fixed

File: components/AutoCompleteInput.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+position:Location;
+indoor:any;
+indoorResults:any;
+lat:number;
+long:number;
+indoorX: number;
+indoorY:number;
+indoorID:string;
+type:string;
+};
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: Add indentation here

File: components/AutoCompleteInput.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+class Autocomplete extends Component<{myposition:Location,type:string},autoStates> {
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: type for prop should be declared above (same as state)

File: components/AutoCompleteInput.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  }
+};
+  async onChangeQuery(query:string){
+  
+    const { position, indoor}=this.state;
+  
+  this.setState({query:query});
+    //add key
+  let key = ""
+  const api = `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${query}&radius=1000&strictbounds&location=${position.getLatitude()},${position.getLongitude()}&key=${key}`;
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: Import the key from your .env file

File: components/AutoCompleteInput.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  const results = await fetch(api);
+  const json = await results.json();
+  this.setState({places:json.predictions});
+
+
+if (query.length===0){
+  this.setState({indoorResults:[]});
+}
+else{
+const filteredData = indoor.filter(x => String(x.identifier.toLowerCase()).includes(query.toLowerCase()));
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: fix indentation

File: components/AutoCompleteInput.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  this.setState({indoorID:sugg.coordinates.y});
+}
+
+ async getOutdoorInfo (sugg:any) {
+  this.setState({query:""});
+  let id = sugg.place_id;
+  
+  
+    //add key
+    let key = ""
```
Comment (razine-bensari) [inline components/AutoCompleteInput.tsx]: Import the key from your .env file

File: App.tsx (hunk 1 above)
Comment (marwan011) [reply App.tsx]: fixed in recent commit

File: components/AutoCompleteInput.tsx (hunk 4 above)
Comment (marwan011) [reply components/AutoCompleteInput.tsx]: resolved

File: components/AutoCompleteInput.tsx (hunk 7 above)
Comment (marwan011) [reply components/AutoCompleteInput.tsx]: resolved

File: components/AutoCompleteInput.tsx (hunk 6 above)
Comment (marwan011) [reply components/AutoCompleteInput.tsx]: resolved

File: components/AutoCompleteInput.tsx (hunk 3 above)
Comment (marwan011) [reply components/AutoCompleteInput.tsx]: resolved

**Pull request merged**